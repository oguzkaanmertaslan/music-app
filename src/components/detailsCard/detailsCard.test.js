import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import DetailsCard, {
  image,
  label,
  listener,
  album,
  playcount,
  alt,
  artist,
} from ".";

test("DetailsCard Text Controls", () => {
  render(<DetailsCard />);
  expect(screen.getByText(/play/i)).toBeInTheDocument();
});
test("DetailsCard with props", () => {
  render(
    <DetailsCard
      image={image}
      label={label}
      listener={listener}
      album={album}
      playcount={playcount}
      alt={alt}
      artist={artist}
    />
  );
});
test("DetailsCard img controls", () => {
  render(<DetailsCard />);
  const image = screen.getByRole("img");
  expect(image).toBeInTheDocument();
  expect(image).toHaveClass("image-area");
});
test("DetailsCard Snapshot", () => {
  const tree = renderer
    .create(
      <DetailsCard
        image={image}
        label={label}
        listener={listener}
        album={album}
        playcount={playcount}
        alt={alt}
        artist={artist}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
