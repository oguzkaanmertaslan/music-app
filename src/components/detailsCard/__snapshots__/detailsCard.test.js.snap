// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsCard Snapshot 1`] = `
<div
  className="card"
>
  <div
    className="left-area"
  >
    <div
      className="image-area"
    >
      <img
        className="image-area"
      />
    </div>
    <div
      className="left-text"
    >
      <br />
    </div>
  </div>
  <div
    className="right-area"
  >
    <div>
       play
    </div>
  </div>
</div>
`;
